### PR TITLE
A C# option to indent/not indent `namespace` contents

### LINKED ISSUE #24129: Add "IndentNamespace" formatting option.
Some people, myself included, prefer to not indent namespace contents. Currently, I don't think the formatting options to enable this are really exposed.

### PR DESCRIPTION
**Pull request by lostmsu**
Added a C# formatting option to enable/disable (enabled by default) indentation for `namespace` contents.

Having this off is very useful for a couple of scenarios:
1. small screens, such as laptop screens
2. diff viewing, when two files have to be shown side by side (GitHub for example often has to wrap C# code because of long lines)

For example, with option off:
```csharp
namespace MyNamespace {

using System;

class MyClass
{
    int f;
}
}
```

Option on:
```csharp
namespace MyNamespace {

  using System;

  class MyClass
  {
    int f;
  }
}
```

Closes #24129

### CODE REVIEW

**Comment by lostmsu** [comment]
Should I regenerate .XLFs and resubmit?

**Comment by CyrusNajmabadi** [comment]
I think this would fall under: https://github.com/dotnet/roslyn/issues/31691

I don't believe it is palatable to add more options like this into the core formatter itself.  Instead, the work should be done to open up the formatter to allow others to add these sorts of one-off formatting options if they so wish.

File: src/VisualStudio/CSharp/Impl/Options/Formatting/IndentationViewModel.cs [hunk 1]
```diff
@@ -61,6 +61,16 @@ class MyClass
     }
 }";
 
+        private const string NamespacePreview = @"
+//[
+namespace MyNamespace {
+    class MyClass
+    {
+    }
+
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/IndentationViewModel.cs]:
```suggestion
```

**Comment by sharwell** [comment]
📝 Marking as blocked on the design review of #24129

File: src/Workspaces/CSharpTest/Formatting/FormattingTests.cs [hunk 2]
```diff
@@ -1427,6 +1427,64 @@ void M()
 }", changedOptionSet: changingOptions);
         }
 
+        [Fact]
+        public async Task IndentNamespaceOn()
+        {
+            var changingOptions = new Dictionary<OptionKey, object>
+            {
+                {  IndentNamespace, true },
+            };
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharpTest/Formatting/FormattingTests.cs]: as this should be the defualt, perhaps do not have these explicit options.

File: src/Workspaces/CSharpTest/Formatting/FormattingTests.cs [hunk 3]
```diff
@@ -1427,6 +1427,64 @@ void M()
...
+expected:
+@"namespace MyNamespace
+{
+using System;
+class MyClass
+{
+    int f;
+}
+}", changedOptionSet: changingOptions);
+        }
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharpTest/Formatting/FormattingTests.cs]: consider additional tests showing what happens with `A.B.C` namespaces as well as nested-namespace scenarios.

**Comment by lostmsu** [comment]
I'll refrain from making any changes to this pull request until the decision on the feature is made.

**Comment by sharwell** [comment]
@lostmsu Thanks! I have it in our high-priority section for Monday.

**Comment by jinujoseph** [comment]
**Design Review Meeting Notes**

The scenario doesn't sound compelling to add more options to core formatter.  We are going to prioritize making the formatter  API to allow others to add these sorts of one-off formatting tracked here https://github.com/dotnet/roslyn/issues/31691

**Pull request closed without merging**